Dockerfile

```diff
@@ -39,12 +39,19 @@ ENV PATH "$PATH:/home/$USER/.local/bin"
 COPY requirements.txt requirements.txt
 RUN pip install -r requirements.txt
 
-# copy source files
+# copy source files (as root)
 COPY gunicorn.conf.py gunicorn.conf.py
 COPY manage.py manage.py
 COPY benefits/ benefits/
 COPY bin/ bin/
 
+# ensure $USER can compile messages in the locale directories
+USER root
+RUN chmod -R 777 /home/$USER/app/benefits/locale
```

Comment:
This change is definitely the one we need to be able to compile the localization messages!!

Confirmed by rebuilding the app container, and entering a `bash` prompt in the container and running the command to compile messages:

![image](https://user-images.githubusercontent.com/1783439/140430743-d6e53a45-2abd-411d-ad70-572579fa8dfd.png)

I wonder if we can find a way to clean this Dockerfile up though? Trying to avoid switching back and forth between `USER` while maintaining a good command ordering in terms of layer cachability.

I think all of these can go _before_ the change to the `calitp` user:

```
# enter app directory
WORKDIR /home/$USER/app

# copy config files
COPY gunicorn.conf.py gunicorn.conf.py
COPY manage.py manage.py

# overwrite default nginx.conf
COPY nginx.conf /etc/nginx/nginx.conf

# copy source files
COPY benefits/ benefits/
COPY bin/ bin/

# ensure $USER can compile messages in the locale directories
RUN chmod -R 777 /home/$USER/app/benefits/locale
```

Then switch, set `PATH`, copy/install requirements, and set entrypoint/command.

My guess is it is likely that the python dependencies change more frequently (e.g. a version update) than the app code at this point, so moving the `pip install` towards the bottom could actually improve our layer caching story?

Reply:
Thanks for that feedback! That all sounds correct to me, and I incorporated those changes.